app/com/microsoft/azure/iotsolutions/devicetelemetry/webservice/v1/controllers/RulesController.java

```
@@ -46,25 +89,96 @@ public Result get(String id) {
      *
      * @return newly created rule
      */
-    public Result post() {
-        return ok(toJson(new RuleApiModel(this.rules.post(new RuleServiceModel()))));
+    public CompletionStage<Result> post() {
+
+        JsonNode jsonBody = request().body().asJson();
+
+        if (jsonBody == null) {
+            log.warn("Error retrieving post request body");
+            return null;
+        }
+
+        ArrayList<ConditionServiceModel> conditions = new ArrayList<>();
+        for (JsonNode condition : jsonBody.withArray("Conditions")) {
+            conditions.add(Json.fromJson(
+                condition,
+                ConditionApiModel.class).toServiceModel());
+        }
+
+        RuleServiceModel ruleServiceModel = new RuleServiceModel(
+            jsonBody.findValue("Name").asText(),
+            jsonBody.findValue("Enabled").asBoolean(),
+            jsonBody.findValue("Description").asText(),
+            jsonBody.findValue("GroupId").asText(),
+            jsonBody.findValue("Severity").asText(),
+            conditions);
+
+        return rules.postAsync(ruleServiceModel)
+            .thenApply(newRule -> {
+                if (newRule != null) {
```

postAsync should never return null. In case of failure it should throw an exception